internal/session.go

```diff
@@ -441,6 +441,14 @@ func sessionCreationActivity(ctx context.Context, sessionID string) error {
 		select {
 		case <-ctx.Done():
 			sessionEnv.CompleteSession(sessionID)
+			// Because of how session creation configures retryPolicy, we need to wrap context cancels that don't
+			// originate from the server as non-retryable errors. See retrypolicy in createSession() above.
+			if !(ctx.Err() == context.Canceled && errors.Is(context.Cause(ctx), &CanceledError{})) {
+				return NewApplicationErrorWithOptions(
+					"context canceled due to worker shutdown", "sessionCreationActivity",
```

Comment:
Nit: I don't think `sessionCreationActivity` is a very clear failure type. It is just repeating the activity type. Maybe something like `SessionWorkerShutdown`. I would also clarify in the error message that the session is failing because the worker is shutdown.